Mathlib/Algebra/Ring/Parity.lean

```diff
@@ -134,11 +134,10 @@ lemma Odd.natCast {R : Type*} [Semiring R] {n : ℕ} (hn : Odd n) : Odd (n : R)
   rw [mul_add, add_mul, mul_one, ← add_assoc, one_mul, mul_assoc, ← mul_add, ← mul_add, ← mul_assoc,
     ← Nat.cast_two, ← Nat.cast_comm]
 
-lemma Odd.pow (ha : Odd a) : ∀ {n : ℕ}, Odd (a ^ n)
-  | 0 => by
-    rw [pow_zero]
-    exact odd_one
-  | n + 1 => by rw [pow_succ]; exact ha.pow.mul ha
+lemma Odd.pow {n : ℕ} (ha : Odd a) : Odd (a ^ n) := by
+  induction n with
+  | zero => simp [pow_zero]
+  | succ n hrec => rw [pow_succ]; exact hrec.mul ha
```

Comment:
Is there a good reason to change the argument order? And the original proof is fine, just change the first line to
` | 0 => by simp [pow_zero]` if you want to golf it ...

Reply:
Yes : this is the argument order of the initial other lemma.

Reply:
I see that the proof of `odd_pow_iff` requires this order (otherwise you have to write `(·.pow)`).